### PR TITLE
#662 Issue: Member type filter should not filter user on connection tab

### LINKED ISSUE #662: Issue: Member type filter should not filter user on connection tab
When we compare result of My Connections tabs on member directory and connection tab on user profile of login use it shows different user due to member type filter. 
Check Screenshots for more understanding. 

**To Reproduce**
1. Login with valid user
2. Open "My Connections" tab on member directory page
3. Open login user "connection" tab 
4 Compare result of both screen

**Expected behavior**
On "My Connections" tabs member should not be hide by member type

**Screenshots**
image

image

### PR DESCRIPTION
**Pull request by chetansatasiya**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #662

THEME PR https://github.com/buddyboss/buddyboss-theme/pull/515

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-members/bp-members-functions.php [hunk 1]
```diff
@@ -3499,26 +3499,33 @@ function bp_register_active_member_types() {
...
+	$args = wp_parse_args( $qs );
 
-	if ( $object != 'members' ) {
+	if ( $object !== 'members' ) {
 		return $qs;
 	}
 
-	if ( bp_is_members_directory() ) {
-		$args = wp_parse_args( $qs );
+	if ( bp_is_members_directory() && isset( $args['scope'] ) && 'all' !== $args['scope'] ) {
```
Comment (mehulkaklotar) [inline src/bp-members/bp-members-functions.php]:
```suggestion
	if ( bp_is_members_directory() && isset( $args['scope'] ) && 'all' === $args['scope'] ) {
```

Comment (chetansatasiya) [reply src/bp-members/bp-members-functions.php]: done

**Comment by chetansatasiya** [comment]
@mehulkaklotar @ankitguptaindia 

I made the changes and deprecated 3 functions `bp_member_type_exclude_users_from_directory_and_searches` , `bp_fixed_all_member_type_count` and `bp_member_type_query`

We need to test below things in this PR to make sure everything work. 

> Members

1 ) Member Directory 
2) Member Directory My Connection
3) Member Directory Member Type Filters
4) Member Type Shortcodes
5) User Profile My Connections, Requests, Following & Mutual Connections

> Groups

1 ) Groups Directory 
2) Groups Directory My Groups
3) Groups Directory Groups Type Filters
4) Groups Type Shortcodes
5) User Profile My Groups, Requests
6) Backend Groups Listing page

**Comment by chetansatasiya** [comment]
@mehulkaklotar I tested and seems working fine.

@ankitguptaindia Can you test this one https://github.com/buddyboss/buddyboss-platform/pull/665#issuecomment-603652330 and let us know if you found anything?

**Comment by ankitguptaindia** [comment]
@chetansatasiya  I found 2 issues here-

1. Members are appearing on member Directory page although that profile type is having setting  ``Member Directory Hidden Setting ON`` https://prnt.sc/rscjc6  https://prnt.sc/rsckc5

2.  After recent fixes, Profile Types which has `Member Directory Hidden Setting ON` not appearing on  User extended Profile edit page.  https://prnt.sc/rscf6b  https://prnt.sc/rscgxn https://ibb.co/RcDtVxz

See attached video for More details :- 

Video Explanation -
https://www.loom.com/share/ff9105fe356c4982b1fcda82d02679d4

**Comment by ankitguptaindia** [comment]
@chetansatasiya  I found 2 issues here-

1. Members are appearing on member Directory page although that profile type is having setting  ``Member Directory Hidden Setting ON`` https://prnt.sc/rscjc6  https://prnt.sc/rsckc5

2.  After recent fixes, Profile Types which has `Member Directory Hidden Setting ON` not appearing on  User extended Profile edit page.  (For admin as well) https://prnt.sc/rscf6b  https://prnt.sc/rscgxn https://ibb.co/RcDtVxz

See attached video for More details :- 

Video Explanation -
https://www.loom.com/share/ff9105fe356c4982b1fcda82d02679d4

**Comment by chetansatasiya** [comment]
> @chetansatasiya I found 2 issues here-
> 
> 1. Members are appearing on member Directory page although that profile type is having setting  `Member Directory Hidden Setting ON` https://prnt.sc/rscjc6  https://prnt.sc/rsckc5. https://prnt.sc/rsd8xl
> 2. After recent fixes, Profile Types which has `Member Directory Hidden Setting ON` not appearing on  User extended Profile edit page.  (For admin as well) https://prnt.sc/rscf6b  https://prnt.sc/rscgxn https://ibb.co/RcDtVxz
> 
> See attached video for More details :-
> 
> Video Explanation -
> https://www.loom.com/share/ff9105fe356c4982b1fcda82d02679d4

@ankitguptaindia 

For the issue 1 this is intentional like we will hide only members/groups only on `All Members` OR `All Groups` tabs only not in `My Connection` OR `My Groups` tabs.

For the issue 2 I fixed the issue.

**Comment by ankitguptaindia** [comment]
@chetansatasiya 

Issues-

Filters are not working on Members-> All Members. Always giving default listing.
https://www.loom.com/share/733f75dfba314302bef2a0aaa88fac44

---

When using Shortcode. Getting some Warning related to Profiles Fileds. https://prnt.sc/rsi1c3

`Notice: Trying to access array offset on value of type bool in /Users/ank/Local Sites/my-qa/app/public/wp-content/plugins/buddyboss-platform/src/bp-core/classes/class-bp-user-query.php on line 709`

----

After Recent fixes. Connection Request page having Filter for Profiles Types. Previously there were not any filter avaiable on Request Page.  https://prnt.sc/rsi7v2

https://www.loom.com/share/4605d1e34667445186fedbc445e69a31
 
----
**Mutual Connection-** Sorting is appearing on Mutual Connection Tab. It should not be there.
https://prnt.sc/rsiiyk



https://github.com/buddyboss/buddyboss-platform/pull/665



----

QUESTIoN RELATED TO Logic implemented previously - Related to Group Types & Profile Types visibility. Options provided there are technically not correct.
https://prnt.sc/rsqhsy

Suggestion- `Display this group type in "Types" filter in Groups Directory ` should auto disable when admin selecting `Hide all groups of this type from Groups Directory `

https://www.loom.com/share/9f792de3ee6b4012bef469339e3d658a

Feel free to clear if I am missing anything.

**Comment by chetansatasiya** [comment]
@ankitguptaindia 

Issues-

Filters are not working on Members-> All Members. Always giving default listing.
https://www.loom.com/share/733f75dfba314302bef2a0aaa88fac44

This is Fixed now 👍 

---

When using Shortcode. Getting some Warning related to Profiles Fileds. https://prnt.sc/rsi1c3

`Notice: Trying to access array offset on value of type bool in /Users/ank/Local Sites/my-qa/app/public/wp-content/plugins/buddyboss-platform/src/bp-core/classes/class-bp-user-query.php on line 709`

This is Fixed now 👍

----

After Recent fixes. Connection Request page having Filter for Profiles Types. Previously there were not any filter avaiable on Request Page.  https://prnt.sc/rsi7v2

https://www.loom.com/share/4605d1e34667445186fedbc445e69a31

This is Fixed now 👍
 
----
**Mutual Connection-** Sorting is appearing on Mutual Connection Tab. It should not be there.
https://prnt.sc/rsiiyk



https://github.com/buddyboss/buddyboss-platform/pull/665


This is Fixed now 👍

----

QUESTIoN RELATED TO Logic implemented previously - Related to Group Types & Profile Types visibility. Options provided there are technically not correct.
https://prnt.sc/rsqhsy

Suggestion- `Display this group type in "Types" filter in Groups Directory ` should auto disable when admin selecting `Hide all groups of this type from Groups Directory `

https://www.loom.com/share/9f792de3ee6b4012bef469339e3d658a

Feel free to clear if I am missing anything.

@ankitguptaindia Now if both checkbox is selected then we will not show that member/group type on main directory pages so it will fix the issues. But if both checkbox selected and if you visit user connections tab in profile page then it will show that member/group type on filter. Let me know your thoughts.

-----

**Comment by eisenwasser** [comment]
@chetansatasiya  

There are too many code changes in this for me to test this and approve.
I need @mehulkaklotar to review, approve and merge.

**Pull request merged**